src/pk-main.c

```diff
@@ -183,6 +183,11 @@ main (int argc, char *argv[])
 
 	/* after how long do we timeout? */
 	exit_idle_time = g_key_file_get_integer (conf, "Daemon", "ShutdownTimeout", NULL);
+        /* THIS COMMENT IS A TSUNAMI STONE 
+         * Before removing the default timeout, please study the git history and
+         * be sure that you are not regressing Redhat bugzilla #1354074 (again). */
+	if (exit_idle_time == 0)
```

Comment:
I wonder if this should look like this:
```suggestion
	if (exit_idle_time < 0)
		exit_idle_time = 0;
	else if (exit_idle_time == 0)
		exit_idle_time = 300;
```
With https://github.com/PackageKit/PackageKit/blob/main/etc/PackageKit.conf#L17 being changed to say `Shut down the daemon after this many seconds idle. -1 means don't shutdown.`

That is because 0 is returned *both* if the `ShutdownTimeout` setting is set to 0, and if it isn't set at all, so if the user sets this to 0, we will still time out, which certainly isn't the desired behavior (not a bug in this patch, probably just a historical oversight).

Reply:
Oh, it's for sure a bug in the patch, and trying to fix it I noticed that the comment was indented with tabs.

This whole thing seems terribly convoluted.  Following the propagation of `ShutdownTimeout`, it is read in as `int exit_idle_time`, then implicitly cast to `uint helper.exit_idle_time`, but only if already greater than zero so that's okay, then implicitly rounded up modulo 5 seconds by `g_timeout_add_seconds()`.

With -1 used thas way, it would be

-1 means no timeout  
0 means 300 seconds  
unspecified means 300 seconds  
anything else means itself rounded up modulo 5 seconds.

I haven't used GLib before, but it seems like by handling the error we can have `ShutdownTimeout=0` behave as documented in the config file comment.  Patch to follow.